sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -73,62 +73,80 @@ private static AmqpMessage BuildAmqpBatchFromMessage(IReadOnlyCollection<Service
                 }
             }
 
-            return BuildAmqpBatchFromMessages(batchMessages, firstMessage, forceBatch);
+            return BuildAmqpBatchFromMessages(batchMessages, forceBatch);
+        }
+
+        /// <summary>
+        ///   Builds a batch of <see cref="AmqpMessage" /> from a set of <see cref="ServiceBusMessage" />
+        ///   optionally propagating the custom properties.
+        /// </summary>
+        ///
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
+        /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
+        ///
+        /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
+        ///
+        public virtual AmqpMessage BuildAmqpBatchFromMessages(IReadOnlyCollection<AmqpMessage> source, bool forceBatch)
+        {
+            return BuildAmqpBatchFromMessages(source.ToList<AmqpMessage>(), forceBatch);
         }
 
         /// <summary>
         ///   Builds a batch <see cref="AmqpMessage" /> from a set of <see cref="AmqpMessage" />.
         /// </summary>
         ///
         /// <param name="batchMessages">The set of messages to use as the body of the batch message.</param>
-        /// <param name="firstMessage">The first message being sent in the batch.</param>
         /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
         ///
         /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
         ///
         private static AmqpMessage BuildAmqpBatchFromMessages(
             List<AmqpMessage> batchMessages,
```

Comment:
Can we take `IReadOnlyCollection<AmqpMessage>` here?   That would allow us to remove the overload and avoid allocating a new list copy.